django/thunderstore/repository/cache.py

```diff
@@ -1,26 +1,39 @@
+from django.db.models import QuerySet
+
 from thunderstore.community.consts import PackageListingReviewStatus
 from thunderstore.community.models import PackageListing, Q
 
 
-def get_package_listing_queryset(community_identifier: str):
-    return (
-        PackageListing.objects.active()
-        .exclude(~Q(community__identifier=community_identifier))
-        .exclude(review_status=PackageListingReviewStatus.rejected)
-        .exclude(
-            Q(community__require_package_listing_approval=True)
-            & ~Q(review_status=PackageListingReviewStatus.approved)
-        )
-        .select_related(
-            "package",
-            "package__owner",
-            "package__latest",
-        )
-        .prefetch_related(
-            "package__versions",
-            "package__versions__dependencies",
-        )
-        .order_by(
-            "-package__is_pinned", "package__is_deprecated", "-package__date_updated"
+def prefetch_package_listing_queryset(
```

Comment:
I assume these are implemented as functions rather than custom QuerySet methods because they're specific to caching and not much use in other cases? I was just wondering if the current function/file name combo is enough to tell others that these probably shouldn't be used for other than caching? Not that I have any perfect suggestions to offer on how to improve it.

Reply:
Yeah, it was originally split like this due to it being easier to explicitly cache if needed. Now there's no explicit cache decorator anymore, but it relies on `django-cachalot` knowing how to cache identical queries.